### DISCUSSION TITLE
Sonoff Touch EU Migration path

### DISCUSSION THREAD

**Discussion by monocycler** [arendst/Tasmota#16397] [Q&A]
Hi
I have one of these I flashed with Tasmota 5.12.0b many years ago : https://templates.blakadder.com/sonoff_touch_eu.html
I just turned it on with the intention of adding to Home Assistant. But the version seem too old to be discovered by the Tasmota integration of Home assistant. So I'd like to migrate it to a more recent version of Tasmota.  https://github.com/arendst/Tasmota/releases says the migration path is to:

- Migrate to Sonoff-Tasmota 5.14
- Migrate to Sonoff-Tasmota 6.7.1 (http://ota.tasmota.com/tasmota/release_6.7.1/sonoff.bin) - NOTICE underscore as a dash is not supported in older versions

But the link to the 5.14 firmware is missing. The binary on  Github is not there, I think. I tried the Wayback Machine e.g. https://web.archive.org/web/20210125153907/https://github.com/arendst/Tasmota/ but no binaries there...
Help! I'd like to avoid messing with Arduino IDE and jumper cable to do the flashing if possible.
Thanks

**Comment by arendst** [comment]
I just made 5.14.0 also available from the OTA website at http://ota.tasmota.com/tasmota/release_5.14.0/

**Reply by sfromis** [reply]
The 5.14 firmware is also available on Github, but of course having it on the OTA site is more convenient for upgrades, instead of having to download from https://github.com/arendst/Tasmota/releases/tag/v5.14.0

**Reply by monocycler** [reply]
Thank you both. I upgraded to 5.14 (via minimal) with success. Then I overstretched and tried to go to 6.7.1-minimal but now it does not connect to my Wifi network nor make an access point :-(
I think I might have to (?compile and?) flash it myself with jumper wires. In this case, what is the best version to use for Sonoff Touch EU (with 2016 printed on the PCB)? (Sonoff Touch is based on the ESP8285. Though the actual chip inside may be a [PSF-A85](https://www.itead.cc/wiki/PSF-A85).)

**Reply by barbudor** [reply]
Where did you get the idea to use `-minimal` ?
There's a warning in **bold characters**:
![image](https://user-images.githubusercontent.com/2996042/187734862-4846dc61-84f3-4018-b986-8d373beb994b.png)

THe link provided for 6.7.1 is : http://ota.tasmota.com/tasmota/release-6.7.1/sonoff.bin

If you go for serial flashing, just don't loose time and go straight with a full-erase, and use the latest version
You will be good for a reconfiguration from scratch but that is straight forward
http://ota.tasmota.com/tasmota/tasmota.bin for serial flashing (dev version)
http://ota.tasmota.com/tasmota/release/tasmota.bin serial flashing release version

**Reply by monocycler** [reply]
I tried going from 5.14 to 6.7.1 and got some kind of error (but did not diligently write it down.)
I got the idea or trying to go from 5.14 to 6.7.1 via 6.7.1-minimal because previously migrating to 5.14 had failed but going to 5.14.0-minimal worked and after that going to 5.14.0 had worked. 
I can't rule having messed up however. I do see in Mosquitto that my Sonoff Touch was last running the full version of 5.14 and not the minimal.
Thanks for the advice about using the latest version. I have an FTDI device and will try to use that with webinstaller. I did manage the 5.12.0b serial flash 5 years ago, so it should work.

**Reply by sfromis** [reply]
In cases where using minimal is relevant (preferable to not use it), never go from one minimal to another, but always to a full binary after each minimal.

**Reply by monocycler** [reply]
I flashed with the stable latest binary and all is well. Thank you all.
